src/Streamly/Internal/Data/Unfold.hs

```diff
@@ -8,6 +8,13 @@
 -- Stability   : experimental
 -- Portability : GHC
 --
+-- The 'Unfold' abstraction is less powerful but more flexible than the Stream
+-- abstraction.  Stream is an unfold with an embedded state. Therefore, Stream
+-- is like an object whereas unfold is like a function.  Unfold is a 'Category'
+-- and an 'Arrow' but not a 'Monad'. Whereas Stream is not a Category or Arrow
```

Comment:
Unfold is a Monad now. Do you want to change the wording?

Reply:
Doing that right now.